packages/ecs/src/UUIDComponent.ts

```
@@ -227,7 +237,11 @@ export const UUIDComponent = defineComponent({
   use: (entity: Entity) => UUIDComponent.join(useComponent(entity, UUIDComponent).value),
 
   /** Joins an EntityUUIDPair into a string */
-  join: (idPair: EntityUUIDPair) => `${idPair.entitySourceID}${idPair.entityID}` as EntityUUID,
+  join: (idPair: EntityUUIDPair) => {
```

These checks seem odd, we should be relying on tyoe safety here, if the value passed in is not valid, we should find what situation is causing this, rather than putting these early returns in. This will likely cause silent failures that will make it harder to find when assumptions are being made.

---

done 4a9d4f728f546766e8ab13ec65283a469d058f54